In [1]:
import os
import subprocess
from subprocess import Popen, PIPE
import glob
from tqdm import tqdm 

In [2]:
source_folder = 'C:/Users/crist/Desktop/TEMIS-AI/1_WebScrapping/v2/DOCUMENT_PROCESSING/DOCUMENTS'
target_folder = 'C:/Users/crist/Desktop/TEMIS-AI/1_WebScrapping/v2/DOCUMENT_PROCESSING/TXT'

In [3]:
files_path = glob.glob('C:/Users/crist/Desktop/TEMIS-AI/1_WebScrapping/v2/DOCUMENT_PROCESSING/DOCUMENTS/*', 
                   recursive = True)
files = []

for file_path in tqdm(files_path):
    file = file_path.replace('C:/Users/crist/Desktop/TEMIS-AI/1_WebScrapping/v2/DOCUMENT_PROCESSING/DOCUMENTS\\','')
    file = file.replace('.rtf','')
    files.append(file)
    
    

100%|██████████| 35737/35737 [00:00<00:00, 744500.83it/s]


In [4]:
files[0:5]

['A001-00', 'A001-01', 'A001-02', 'A001-03', 'A001-04']

1. files[0:4467]
2. files[4467:8934]
3. files[8934:13401]
4. files[13401:17868]
5. files[17868:22335]
6. files[22335:26802]
7. files[26802:31269]
8. files[31269:35737]

In [5]:
len(files)

35737

In [6]:
files[35736]

'T-999A-01'

In [ ]:
for file in tqdm(files[8934:13401]):
    
    file_name = file
    command = 'docto -f ' + source_folder + '/' + file_name + '.rtf' + ' -O "' + target_folder + '/' + file_name + '.txt" -T wdFormatText' 
    #print(command)
    process = Popen(command, stdout=PIPE, stderr=PIPE)
    stdout, stderr = process.communicate()
    #print(stdout)


 42%|████▏     | 1877/4467 [3:44:23<4:13:03,  5.86s/it]  

In [7]:
for file in tqdm(files[10810:13401]):
    
    file_name = file
    command = 'docto -f ' + source_folder + '/' + file_name + '.rtf' + ' -O "' + target_folder + '/' + file_name + '.txt" -T wdFormatText' 
    #print(command)
    process = Popen(command, stdout=PIPE, stderr=PIPE)
    stdout, stderr = process.communicate()
    #print(stdout)

 66%|██████▌   | 1700/2591 [2:08:22<1:07:16,  4.53s/it] 


KeyboardInterrupt: 

In [9]:
for file in tqdm(files[12509:13401]):
    
    file_name = file
    command = 'docto -f ' + source_folder + '/' + file_name + '.rtf' + ' -O "' + target_folder + '/' + file_name + '.txt" -T wdFormatText' 
    #print(command)
    process = Popen(command, stdout=PIPE, stderr=PIPE)
    stdout, stderr = process.communicate()
    #print(stdout)

100%|██████████| 892/892 [1:05:24<00:00,  4.40s/it]
